### PR TITLE
admin, restriction area

### PR DESCRIPTION
**Pull request by cmangeat**

### CODE REVIEW

File: admin/acceptance_tests/restriction_areas_test.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import re
+from geoalchemy2.shape import from_shape
+from shapely.geometry import box, Polygon, shape
+
+from . import AbstractViewsTests
+
+
+@pytest.fixture(scope='class')
+@pytest.mark.usefixtures('dbsession')
+def restriction_area_test_data(dbsession):
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/restriction_areas_test.py]: You can already set scope to 'function' and add 'transact' in usefixture.
Note that you need to add trasact also as parameter and del transact for linter.
In that case you should remove internal transaction.
I would prefer having a flush before return.

File: admin/c2cgeoportal_admin/schemas/roles.py [hunk 2]
```diff
@@ -6,19 +6,27 @@
...
-        edit_url=lambda request, value: request.route_url(
-            'c2cgeoform_item',
-            table='roles',
-            id=value
-        )
-    ),
-    validator=manytomany_validator
-)
+
+def build(name):
```
Comment (arnaud-morvan) [inline admin/c2cgeoportal_admin/schemas/roles.py]: Is it possible to give an explicit name, ex : `def roles_schema_node(name):`

File: admin/c2cgeoportal_admin/schemas/roles.py [hunk 3]
```diff
@@ -6,19 +6,27 @@
...
+                table='roles',
+                id=value
+            )
+        ),
+        validator=manytomany_validator,
+        missing=colander.drop
+    )
+
+
+roles_schema_node_for_theme = build('restricted_roles')
```
Comment (arnaud-morvan) [inline admin/c2cgeoportal_admin/schemas/roles.py]: I would prefer to call this directly in parent schema contruction in views modules.

Comment (cmangeat) [reply admin/c2cgeoportal_admin/schemas/roles.py]: We were used to clone nodes in view... Is there a difference between scratch building and cloning ?

**Pull request merged**